Add "main.h" file not found to Coding FAQ (#129)
purduesigbots/pros-docs: v5/tutorials/general/coding-faq.rst

Coding FAQ
==========

    * ``"main.h" file not found``:

>         The linter being separate from the compiler produces this error when

Reviewer: Not sure how important it is to mention that the linter is separate from the compiler here

```suggestion
This error can occur when the compilation cache
```

Coding FAQ
==========

    * ``"main.h" file not found``:
        This error occurs when the compilation database ``compile_commands.json``

>         is not up to date. The file contains the exact compiler calls of the project.

Reviewer: ```suggestion
        is not up to date. The file contains the exact compiler calls of the project, and should be automatically populated when the code is compiled.
```

>         Right after project creation, this could occur because the project was created

Reviewer: ```suggestion
        If you see this error just after creating a project, it could be because the project was created
```

        using the CLI without the ``–compile-after flag``, or there was a hang-up in the

>         editor after creating the project. By running ``pros make all`` or in the editor

Reviewer: ```suggestion
        editor after creating the project. Regardless of when this issue appears, running ``pros make all`` in the CLI or
```

clarify that this should fix it regardless of when it happens

>         running ``PROS -> Build -> All``, the whole project is compiled again, fixing this

Reviewer: ```suggestion
        running ``PROS -> Build -> All`` in the editor, the whole project is compiled again, fixing this
```

>         issue. It is possible that a compilation error was not caught by the linter, so running

Reviewer: ```suggestion
        issue. You may need to close and reopen any files in order for the linter to catch up. It is also possible that a compilation error was not caught by the linter, so running
```

==========

    * ``"main.h" file not found``:
        This error occurs when the compilation database ``compile_commands.json``
        is not up to date. The file contains the exact compiler calls of the project.
        Right after project creation, this could occur because the project was created

>         using the CLI without the ``–compile-after flag``, or there was a hang-up in the

Reviewer: take "flag" out of the code fence
